lib/osf-components/addon/components/item-metadata-manager/component.ts

```diff
@@ -0,0 +1,109 @@
+import Store from '@ember-data/store';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { alias, or } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+import { BufferedChangeset } from 'ember-changeset/types';
+import { restartableTask, task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+
+import CustomItemMetadataRecordModel from 'ember-osf-web/models/custom-item-metadata-record';
+import NodeModel from 'ember-osf-web/models/node';
+import { Permission } from 'ember-osf-web/models/osf-model';
+import RegistrationModel from 'ember-osf-web/models/registration';
+import { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
+
+interface Args {
+    node: (NodeModel | RegistrationModel);
```

Comment:
Similar comment here. Could this just be `AbstractNodeModel`, or is it more informative to explicitly say it's either a Node or a Registration?

Reply:
If we had typing in templates, it would be more important, but for what we have, this makes sense.